src/Features/Core/Portable/ReplaceMethodWithProperty/ReplaceMethodWithPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,417 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Rename;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.ReplaceMethodWithProperty
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, LanguageNames.VisualBasic,
+        Name = nameof(ReplaceMethodWithPropertyCodeRefactoringProvider)), Shared]
+    internal class ReplaceMethodWithPropertyCodeRefactoringProvider : CodeRefactoringProvider
+    {
+        private const string GetPrefix = "Get";
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            var service = document.GetLanguageService<IReplaceMethodWithPropertyService>();
+            if (service == null)
+            {
+                return;
+            }
+
+            var cancellationToken = context.CancellationToken;
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var position = context.Span.Start;
+            var token = root.FindToken(position);
+
+            if (!token.Span.Contains(context.Span))
+            {
+                return;
+            }
+
+            var methodDeclaration = service.GetMethodDeclaration(token);
+            if (methodDeclaration == null)
+            {
+                return;
+            }
+
+            // Ok, we're in the signature of the method.  Now see if the method is viable to be 
+            // replaced with a property.
+            var methodName = service.GetMethodName(methodDeclaration);
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var methodSymbol = semanticModel.GetDeclaredSymbol(methodDeclaration) as IMethodSymbol;
+            if (!IsValidGetMethod(methodSymbol))
+            {
+                return;
+            }
+
+            var hasGetPrefix = HasGetPrefix(methodName);
```

Comment:
then it won't have the "Get" prefix and we won't remove it.  I'm skeptical that pattern is important enough to look for.